Makefile

```diff
@@ -236,13 +236,15 @@ ifdef IS_LINUX
 LDFLAGS += -Wl,--no-undefined
 endif
 
+CFLAGS += -O2 -fno-omit-frame-pointer -g
+
 ifdef COVERAGE
-	CFLAGS += -g -O0 --coverage
+	CFLAGS += -O0 --coverage
```

Comment:
`-O0` will override `-O2` from outer scope?

Reply:
Yeah. The compiler looks at the last specified level. With DEBUG enabled we end up with `-O2 -O0` which is equivalent to just `-O0`. (Individual flags like `-fno-omit-frame-pointer` are effective regardless of the optimization level set by -O.)